portal/lib/track_portal_events.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * track_portal_events.php
+ *
+ * @package        OpenEMR
+ * @link           https://www.open-emr.org
+ * @author         Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright      Copyright (c) 2025 <sjpadgett@gmail.com>
+ * @license        https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once(__DIR__ . '/../../src/Common/Session/SessionUtil.php');
+OpenEMR\Common\Session\SessionUtil::portalSessionStart();
+
+$sessionAllowWrite = true;
+if (isset($_SESSION['pid']) && isset($_SESSION['patient_portal_onsite_two'])) {
+    $pid = $_SESSION['pid'];
+    $ignoreAuth_onsite_portal = true;
+    require_once(__DIR__ . '/../../interface/globals.php');
+} else {
+    OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
+    $ignoreAuth = false;
+    require_once(__DIR__ . '/../../interface/globals.php');
+    if (!isset($_SESSION['authUserID'])) {
+        $landingpage = 'index.php';
+        header('Location: ' . $landingpage);
+        exit;
+    }
+}
+
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Services\VersionService;
+use OpenEMR\Telemetry\TelemetryRepository;
+use OpenEMR\Telemetry\TelemetryService;
+
+header("Content-Type: application/json");
+
+/**
+ * Main request handler that reads input, verifies the CSRF token, and delegates
+ * to the appropriate telemetry service method.
+ */
+function handleRequest(): void
+{
+    // Read JSON payload.
+    $input_json = file_get_contents('php://input');
+    $data = json_decode($input_json, true);
+
+    // Verify CSRF token.
+    if (!isset($data["csrf_token_form"]) || !CsrfUtils::verifyCsrfToken($data["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+
+    $telemetryRepo = new TelemetryRepository();
+    $versionService = new VersionService();
+    $telemetryService = new TelemetryService($telemetryRepo, $versionService);
+
+    $action = $data['action'] ?? '';
+    switch ($action) {
+        case 'portalCardClickData':
+            $telemetryService->reportClickEvent($data);
+            break;
+        case 'allUsageData':
```

Comment:
I can't see where this is being used in the code. If not needed, wouldn't include it, since then giving access to a patient to view all usage data (and probably worse, have it truncated).

Reply:
actually, looks like they won't be able to view since it just kicks off the report to the reporting engine. Still wouldn't allow patient access to do this since would mess up the reporting.

Reply:
yep, didn't need it. I think I left because I may have needed it to kick off reporting.
I'll remove.